bluefog/torch/mpi_win_ops.cc

```diff
@@ -388,6 +388,7 @@ int DoWinPut(::torch::Tensor tensor, const std::string& name,
     ::torch::Tensor cpu_buffer =
         tensor.to(::torch::Device(::torch::kCPU), /*non_blocking=*/false);
     bf_tensor = std::make_shared<TorchTensor>(cpu_buffer);
+    device = CPU_DEVICE_ID;
```

Comment:
Don't quite understand the device change here.

Reply:
See above